app/main.go

```diff
@@ -96,9 +105,62 @@ func main() {
 		log.Fatalf("Invalid flags: %v", err)
 	}
 
-	http.HandleFunc("/", index)
-	http.HandleFunc("/sync", sync)
-	http.HandleFunc("/cleanup", cleanup)
+	config := tsbridge.NewConfig(&tsbridge.ConfigOptions{
+		Filename:                 *metricConfig,
+		MinPointAge:              *minPointAge,
+		CounterResetInterval:     *counterResetInterval,
+		SDLookBackInterval:       *sdLookBackInterval,
+		SDInternalMetricsProject: *sdInternalMetricsProject,
+		UpdateParallelism:        *updateParallelism,
+		EnableStatusPage:         *enableStatusPage,
+		StorageEngine:            *storageEngine,
+		SyncPeriod:               *syncPeriod,
+		SyncCleanupAfter:         *syncCleanupAfter,
+	})
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		index(w, r, config)
+	})
+	http.HandleFunc("/sync", func(w http.ResponseWriter, r *http.Request) {
+		syncHandler(w, r, config)
+	})
+	http.HandleFunc("/cleanup", func(w http.ResponseWriter, r *http.Request) {
+		cleanupHandler(w, r, config)
+	})
+
+	// Run a sync loop for standalone use
+	if env.NotAppEngine() {
+		log.Debug("Running outside of appengine, starting up a sync loop...")
+		ctx, cancel := context.WithCancel(context.Background())
+		count := 0
+		go func() {
+			defer cancel()
+			for {
+				select {
+				case <-time.After(config.Options.SyncPeriod):
+					ctx, cancel := context.WithTimeout(ctx, *updateTimeout)
+					log.WithContext(ctx).Debugf("Running sync...")
+					if err := sync(ctx, config); err != nil {
+						log.WithContext(ctx).Errorf("error running sync() routine: %v", err)
+						return
```

Comment:
What's the strategy behind returning here? It seems like this will stop the sync loop, but the process will keep running.

Reply:
There are sometimes transient errors with running this routine (issues on Datadog or SD side for example).

This is a good point though - should I introduce a backoff counter here? E.g. exit the loop if it failed more than 3 times.

Reply:
I am not sure I understand what would exiting the loop achieve if the process will keep running.

I don't think you need backoff here given how rarely syncing happens (minutes rather than seconds). I think logging an error and trying again during the next scheduled sync is sufficient.

Reply:
Oh ok, no, this is just my mistake as I didn't intend to quit the main loop. Fixed.